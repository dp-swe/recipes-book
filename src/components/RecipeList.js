import React from 'react';
import {connect} from 'react-redux';
import {recipesSelector} from '../store/ducks/recipesDuck';
import RecipeItem from './RecipeItem';
import styles from './styles/recipeList.module.scss';

const RecipesList = ({recipes}) => (
    <div className={styles.recipe_list}>    
        {recipes.map(item => 
                <RecipeItem item={item}/>
            )}
    </div>
)

const mapStateToProps = (state) => ({
    recipes: recipesSelector(state)
})

export default connect(mapStateToProps, null)(RecipesList);

