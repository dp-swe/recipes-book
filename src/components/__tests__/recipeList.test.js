import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { RecipesList } from '../RecipeList';

describe('RecipeList Component', () => {
    let recipes;
    let wrapper;

    beforeEach(() => {
        recipes = [{
                title: "Roasted Pepper and Bacon Omelet",
                href: "http://www.bigoven.com/43919-Roasted-Pepper-and-Bacon-Omelet-recipe.html",
                ingredients: "eggs, salt, black pepper, butter, black pepper, bacon, onions, garlic, roasted red peppers, oregano, black pepper",
                thumbnail: ""
            },
            {
                title: " Broccoli Oven Omelet Recipe ",
                href: "http://cookeatshare.com/recipes/broccoli-oven-omelet-92851",
                ingredients: "eggs, broccoli, onions, parmesan cheese, lowfat milk, salt, basil, garlic, tomato, parmesan cheese",
                thumbnail: ""
            }];
        wrapper = shallow(<RecipesList recipes={recipes}/>);
    })

    it('should render without broken', () => {
        expect(wrapper.exists()).toBe(true);
    })

    it('should render correctly', () => {
        const tree = renderer.create(<RecipesList recipes={recipes}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should contain a list of recipes', () => {
        expect(wrapper.find("li")).toHaveLength(2);
    })
})