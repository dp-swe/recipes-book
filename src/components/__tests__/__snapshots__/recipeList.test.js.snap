// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList Component should render correctly 1`] = `
<ul
  className="recipe_list"
>
  <li
    className="recipe_list__item"
  >
    <a
      className="recipe__a"
      href="http://www.bigoven.com/43919-Roasted-Pepper-and-Bacon-Omelet-recipe.html"
    >
      <div
        className="recipe__image_wrapper"
      >
        <img
          alt="Recipe"
          className="recipe__image"
          src=""
        />
      </div>
      <div
        className="recipe__title"
      >
        Roasted Pepper and Bacon Omelet
      </div>
      <button
        className="recipe__button"
      >
        Make Favorite
      </button>
      <ul
        className="recipe__ingredients"
      >
        <li
          className="recipe__ingredient"
        >
          eggs
        </li>
        <li
          className="recipe__ingredient"
        >
           salt
        </li>
        <li
          className="recipe__ingredient"
        >
           black pepper
        </li>
        <li
          className="recipe__ingredient"
        >
           butter
        </li>
        <li
          className="recipe__ingredient"
        >
           black pepper
        </li>
        <li
          className="recipe__ingredient"
        >
           bacon
        </li>
        <li
          className="recipe__ingredient"
        >
           onions
        </li>
        <li
          className="recipe__ingredient"
        >
           garlic
        </li>
        <li
          className="recipe__ingredient"
        >
           roasted red peppers
        </li>
        <li
          className="recipe__ingredient"
        >
           oregano
        </li>
        <li
          className="recipe__ingredient"
        >
           black pepper
        </li>
      </ul>
    </a>
  </li>
  <li
    className="recipe_list__item"
  >
    <a
      className="recipe__a"
      href="http://cookeatshare.com/recipes/broccoli-oven-omelet-92851"
    >
      <div
        className="recipe__image_wrapper"
      >
        <span
          className="recipe__label"
        >
          Has Lactose
        </span>
        <img
          alt="Recipe"
          className="recipe__image"
          src=""
        />
      </div>
      <div
        className="recipe__title"
      >
         Broccoli Oven Omelet Recipe 
      </div>
      <button
        className="recipe__button"
      >
        Make Favorite
      </button>
      <ul
        className="recipe__ingredients"
      >
        <li
          className="recipe__ingredient"
        >
          eggs
        </li>
        <li
          className="recipe__ingredient"
        >
           broccoli
        </li>
        <li
          className="recipe__ingredient"
        >
           onions
        </li>
        <li
          className="recipe__ingredient"
        >
           parmesan cheese
        </li>
        <li
          className="recipe__ingredient"
        >
           lowfat milk
        </li>
        <li
          className="recipe__ingredient"
        >
           salt
        </li>
        <li
          className="recipe__ingredient"
        >
           basil
        </li>
        <li
          className="recipe__ingredient"
        >
           garlic
        </li>
        <li
          className="recipe__ingredient"
        >
           tomato
        </li>
        <li
          className="recipe__ingredient"
        >
           parmesan cheese
        </li>
      </ul>
    </a>
  </li>
</ul>
`;
